master/internal/trial.go

```diff
@@ -1230,3 +1305,75 @@ func (t *trial) tellWithSnapshot(
 	}))
 	return nil
 }
+
+type (
+	// watchPreemption begins watching if the task has been preempted.
+	// The task responds to this message with a channel of bools, where sends of true
+	// indicate to preempt and sends of false are used to synchronize (e.g. you want to
+	// block until you receive _something_ but not until the first preemption).
+	watchPreemption   struct{ id uuid.UUID }
+	preemptionWatcher struct{ C <-chan struct{} }
+	unwatchPreemption struct{ id uuid.UUID }
+
+	// preemption represents the preemption status of a task. A task is assumed to be preempted
+	// exactly one time. The object is "nil safe" - it'll gracefully handle calls on a nil
+	// preemption. This is nice until we move to trial has many task actors / generic task actor,
+	// where the lifetime of a "preemption" is equivalent to the lifetime of task and they can be
+	// initialized together.
+	preemption struct {
```

Comment:
long term, when push arch is done, i plan to do some refactoring. i will move this somewhere. i was thinking it could go into internal/task or internal/job or something. maybe pkg/task could be renamed pkg/taskspec